### PR TITLE
[Merged by Bors] - feat(Logic/Relation): add missing relation material and show `⩿ = ReflGen (· ⋖ ·)`

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Logic/Relation.lean [hunk 1]
```diff
@@ -576,6 +596,33 @@ theorem reflTransGen_swap : ReflTransGen (swap r) a b ↔ ReflTransGen r b a :=
   ⟨ReflTransGen.swap, ReflTransGen.swap⟩
 #align relation.refl_trans_gen_swap Relation.reflTransGen_swap
 
+lemma reflGen_comp_transGen : ReflGen (TransGen r) = ReflTransGen r := by
```
Comment (eric-wieser) [inline Mathlib/Logic/Relation.lean:599]:
```suggestion
lemma reflGen_transGen : ReflGen (TransGen r) = ReflTransGen r := by
```
There's no `comp` in this expression

Comment (eric-wieser) [reply Mathlib/Logic/Relation.lean:599]: (same throughout)

File: Mathlib/Logic/Relation.lean [hunk 2]
```diff
@@ -448,6 +448,18 @@ theorem _root_.WellFounded.transGen (h : WellFounded r) : WellFounded (TransGen
   ⟨fun a ↦ (h.apply a).TransGen⟩
 #align well_founded.trans_gen WellFounded.transGen
 
+section reflGen
+
+lemma reflGen_eq_self [IsRefl α r] : ReflGen r = r := by
```
Comment (eric-wieser) [inline Mathlib/Logic/Relation.lean:453]: I'm not too happy with how this takes `IsRefl` but a later lemma takes `Reflexive r`. I think convention for this file is to ignore the typeclasses.

Comment (eric-wieser) [reply Mathlib/Logic/Relation.lean:453]:
```suggestion
lemma reflGen_eq_self (hr : IsTransitive r) : ReflGen r = r := by
```

**Review by hrmacbeth — APPROVED** [review]

File: Mathlib/Logic/Relation.lean [hunk 3]
```diff
@@ -576,6 +598,33 @@ theorem reflTransGen_swap : ReflTransGen (swap r) a b ↔ ReflTransGen r b a :=
   ⟨ReflTransGen.swap, ReflTransGen.swap⟩
 #align relation.refl_trans_gen_swap Relation.reflTransGen_swap
 
+lemma reflGen_transGen : ReflGen (TransGen r) = ReflTransGen r := by
```
Comment (ocfnash) [inline Mathlib/Logic/Relation.lean:601]:
```suggestion
@[simp] lemma reflGen_transGen : ReflGen (TransGen r) = ReflTransGen r := by
```

File: Mathlib/Logic/Relation.lean [hunk 4]
```diff
@@ -576,6 +598,33 @@ theorem reflTransGen_swap : ReflTransGen (swap r) a b ↔ ReflTransGen r b a :=
...
+  ext x y
+  simp_rw [reflTransGen_iff_eq_or_transGen, reflGen_iff]
+
+lemma transGen_reflGen : TransGen (ReflGen r) = ReflTransGen r := by
```
Comment (ocfnash) [inline Mathlib/Logic/Relation.lean:605]:
```suggestion
@[simp] lemma transGen_reflGen : TransGen (ReflGen r) = ReflTransGen r := by
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**